packages/nimble-components/src/table/models/tests/sort-operations.spec.ts

```diff
@@ -0,0 +1,99 @@
+import type { TableFieldValue } from '../../types';
+
+describe('Sort Operations', () => {
+    function createFakeRow(
+        value: TableFieldValue
+    ): TanStackRow<TableFieldValue> {
+        const row = jasmine.createSpyObj<TanStackRow<TableFieldValue>>(
+            'TanStackRow',
+            ['getValue']
+        );
+        row.getValue.and.returnValue(value);
+        return row;
+    }
+
+    const basicNumericSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: Number.NEGATIVE_INFINITY, name: '-∞' },
+        { value: Number.MIN_SAFE_INTEGER, name: 'MIN_SAFE_INTEGER' },
+        { value: -9999, name: '-9999' },
+        { value: -3, name: '-3' },
+        { value: 0, name: '0' },
+        { value: Number.EPSILON, name: 'EPSILON' },
+        { value: 3, name: '3' },
+        { value: 9999, name: '9999' },
+        { value: Number.MAX_SAFE_INTEGER, name: 'MAX_SAFE_INTEGER' },
+        { value: Number.POSITIVE_INFINITY, name: '∞' }
+    ] as const;
+    const basicBooleanSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: false, name: 'false' },
+        { value: true, name: 'true' }
+    ] as const;
+    const basicStringSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: '', name: "''" },
+        { value: 'abc', name: 'abc' },
+        { value: 'zzz', name: 'zzz' }
+    ] as const;
+
+    const localeAwareCaseSensitiveSortOrder = [
+        { value: undefined, name: 'undefined' },
+        { value: null, name: 'null' },
+        { value: '', name: "''" },
+        { value: 'abc', name: 'abc' },
+        { value: 'ABC', name: 'ABC' },
+        { value: 'hello world', name: 'hello WORLD' },
+        { value: 'xyz', name: 'xyz' }
+    ];
+
+    const sortOrderTestCases = [
+        {
+            sortOperation: TableColumnSortOperation.basic,
+            testCases: [
+                basicNumericSortOrder,
+                basicBooleanSortOrder,
+                basicStringSortOrder
+            ]
+        },
+        {
+            sortOperation: TableColumnSortOperation.localeAwareCaseSensitive,
+            testCases: [localeAwareCaseSensitiveSortOrder]
+        }
+    ] as const;
+    for (const sortOrderTestCase of sortOrderTestCases) {
+        for (const expectedSortOrder of sortOrderTestCase.testCases) {
+            for (let i = 0; i < expectedSortOrder.length; i++) {
+                for (let j = 0; j < expectedSortOrder.length; j++) {
+                    const sortOrderA = expectedSortOrder[i]!;
+                    const sortOrderB = expectedSortOrder[j]!;
+                    it(`comparing ${sortOrderA.name} to ${sortOrderB.name} returns expected value`, () => {
+                        let expectedSortResult: number;
+                        if (i === j) {
+                            expectedSortResult = 0;
```

Comment:
No action required, just documenting a train of thought I had about our position on [stable sorting](https://en.wikipedia.org/wiki/Category:Stable_sorts) (items with the same value being sorted in a consistent order)...

I was going to suggest that we update each sort order array to include a duplicate value and then verify that they are sorted in a stable order. But I guess since these tests cover the comparison function rather than the result of the sort, that wouldn't be any different than the existing case where `i === j` (and in fact those cases would break the assumption that `array[i] < array[i+1]`).

So to properly test stable sorting we'd need to ensure that equal values return a sort result of 0 (which these tests handle) and then also ensure that the TanStack sort function maintains sort order when the sort result is 0. The latter would presumably be the job of TanStack (or at least a configuration we pass to TanStack) but I'm not sure if they even support stable sort. I found a couple of issues requesting it that were closed with no action.
https://github.com/TanStack/table/issues/2188
https://github.com/TanStack/table/issues/913

Anyway, for the scope of this PR I think we're fine. Just recording this for discussion.